### ISSUE TITLE
ONNX sklearn model interface for android only returns class label as output

### ISSUE THREAD

**Issue by debajyoti94** [microsoft/onnxruntime#14984]
### Describe the issue

[ONNX sklearn model interface for android only returns class label as output](https://stackoverflow.com/questions/75615525/onnx-sklearn-model-interface-for-android-only-returns-class-label-as-output)

I am trying to get probability/confidence score from the model on my android application. While creating the ONNX model in python, I am able to see the confidence/probability score for a given input. However, when I give the same input on Android, the `OrtSession.run()` method returns only the class label.

I tried adding `zipmap` in the `options` while converting sklearn model to onnx but that did not help either.
I do not know if the issue is on Android side or while creating the `.onnx` model. Hence attaching the code of both sides.

### To reproduce

ONNX code
```
# sample classification
from sklearn.datasets import load_iris
from sklearn import tree
# from sklearn.linear_model import LogisticRegression
iris = load_iris()
X, y = iris.data, iris.target
# clf = tree.DecisionTreeClassifier()
clf = tree.DecisionTreeClassifier()≠
clf = clf.fit(X, y)


# Convert into ONNX format
from skl2onnx import convert_sklearn
from skl2onnx.common.data_types import FloatTensorType
import numpy as np
initial_type = [('float_input', FloatTensorType([None, 4]))]
options = {id(clf): {'zipmap': True}}
onx_dt = convert_sklearn(clf, initial_types=initial_type,
                      options=options)

with open("dt_iris.onnx", "wb") as f:
    f.write(onx_dt.SerializeToString())

# Compute the prediction with ONNX Runtime
import onnxruntime as rt
import numpy as np

sess = rt.InferenceSession(onx.SerializeToString())
res = sess.run(None, {'float_input': np.array([[0.9, 0.7, 0.9, 0.4]],
                                              dtype=np.float32)}) #X_test.astype(numpy.float32)
print(res)
print(res[1][:2])
print("probabilities type:", type(res[1]))
print("type for the first observations:", type(res[1][0]))

```
The output for the above classification is

```
[array([1], dtype=int64), [{0: 0.09890109300613403, 1: 0.901098906993866}]]
[{0: 0.09890109300613403, 1: 0.901098906993866}]
probabilities type: <class 'list'>
type for the first observations: <class 'dict'>
```
Android code
```
private fun runPrediction(input: String, ortSession: OrtSession, ortEnvironment: OrtEnvironment): String {
        // Get the name of the input node
        //val value = input.toFloat()
        val value = floatArrayOf(0.6f,0.6f,0.6f,0.6f)
        val inputName = ortSession.inputNames?.iterator()?.next()
        // Make a FloatBuffer of the inputs
        val floatBufferInputs = FloatBuffer.wrap(value)
        // Create input tensor with floatBufferInputs of shape ( 1 , 1 )
        val inputTensor = OnnxTensor.createTensor(ortEnvironment, floatBufferInputs, longArrayOf(1, 4))
        // Run the model
        val options = OrtSession.RunOptions().apply {
            logLevel = OrtLoggingLevel.ORT_LOGGING_LEVEL_VERBOSE
        }
        val results = ortSession.run(mapOf(inputName to inputTensor), options)
        val rawOutput = results[0].value
        return (rawOutput as LongArray).toList().toString()
    }
```

### Urgency

We are trying a POC of deploying a sklearn-based model to Android and we want to set custom thresholds for predicting a certain class/label. This issue is a blocker for us, as we are able to only extract the class label and not the probability on android.

### Platform

Android

### OS Version

13

### ONNX Runtime Installation

Released Package

### Package Name (if 'Released Package')

None

### ONNX Runtime Version or Commit ID

1.14.0

### ONNX Runtime API

Java/Kotlin

### Architecture

ARM64

### Execution Provider

Default CPU

**Comment by Craigacp** [comment]
I think you're accessing the first element of the result in Java, which is the class label, and the second element of the result in Python, which is the class probabilities. If you change `val rawOutput = results[0].value` to `val rawOutput = results[1].value` and then fix the return statement on the next line you should be ok.

**Comment by natke** [comment]
Thank you @Craigacp! @debajyoti94 Can you please try the solution suggested by @Craigacp and let us know if that solved your problem.

**Comment by debajyoti94** [comment]
Accessing the 1st index and casting it to `List<OnnxMap>`worked! Attaching the code below for reference.
Thank you @Craigacp!

```
private fun runPrediction(ortSession: OrtSession, ortEnvironment: OrtEnvironment): String {
        val value = floatArrayOf(0.9f, 0.7f, 0.9f, 0.4f)
        val inputName = ortSession.inputNames?.iterator()?.next()
        val floatBufferInputs = FloatBuffer.wrap(value)
        val inputTensor = OnnxTensor.createTensor(ortEnvironment, floatBufferInputs, longArrayOf(1, 4))
        // Run the model
        val options = OrtSession.RunOptions().apply {
            logLevel = OrtLoggingLevel.ORT_LOGGING_LEVEL_VERBOSE
        }
        val results = ortSession.run(mapOf(inputName to inputTensor), options)
        val rawOutput = results[1].value as List<OnnxMap>
        return (rawOutput[0].value).toString()
    }
```
